src/main/scala/mail/Mail.scala

```diff
@@ -0,0 +1,35 @@
+package io.blindnet.identity
+package mail
+
+import cats.data.EitherT
+import cats.effect.*
+import org.apache.commons.mail.*
+
+import scala.util.Try
+
+case class Mail(
+  from: Sender,
+  to: List[String],
+  subject: String,
+  message: String,
+  richMessage: Option[String] = None,
+) {
+  def send(config: MailConfig): IO[Unit] = IO {
```

Comment:
By fp guidelines, `case class` shouldn't really cause any effects nor depend on services. It is just a simple [ADT](https://blog.rockthejvm.com/algebraic-data-types/) representing a part of the domain.
Change it into `class` or make the `EmailService` (I prefer the latter).